app/lib/import_export.rb

```diff
@@ -378,14 +716,130 @@ def jurisdiction_paths(patients)
-    patients.each do |patient|
-      patients_jurisdiction_names[patient.id] = jurisdiction_names[patient.jurisdiction_id]
+  # Extracts assessment data values given relevant fields
+  def extract_assessments_details(patients_identifiers, assessments, fields)
+    if fields.include?(:symptoms)
+      conditions = ReportedCondition.where(assessment_id: assessments.pluck(:id))
+      symptoms = Symptom.where(condition_id: conditions.pluck(:id)).order(:label)
+
+      conditions_hash = Hash[conditions.pluck(:id, :assessment_id).map { |id, assessment_id| [id, assessment_id] }]
+                        .transform_values { |assessment_id| { assessment_id: assessment_id, symptoms: {} } }
+      symptoms.each do |symptom|
+        conditions_hash[symptom[:condition_id]][:symptoms][symptom[:name]] = symptom.value
+      end
+      assessments_hash = Hash[conditions_hash.map { |_, condition| [condition[:assessment_id], condition[:symptoms]] }]
     end
-    patients_jurisdiction_names
+
+    symptom_names_and_labels = symptoms&.distinct&.pluck(:name, :label)
+
+    assessments_details = []
+    assessments.each do |assessment|
+      assessment_details = {}
+      assessment_details[:patient_id] = assessment[:patient_id] || '' if fields.include?(:patient_id)
+      PATIENT_SECONDARY_IDENTIFIER_FIELDS.each do |identifier|
+        assessment_details[identifier] = patients_identifiers[assessment[:patient_id]][identifier] || '' if fields.include?(identifier)
+      end
+      assessment_details[:id] = assessment[:id] || '' if fields.include?(:id)
+      assessment_details[:symptomatic] = assessment[:symptomatic] || false if fields.include?(:symptomatic)
+      assessment_details[:who_reported] = assessment[:who_reported] || '' if fields.include?(:who_reported)
+      assessment_details[:created_at] = assessment[:created_at] || '' if fields.include?(:created_at)
+      assessment_details[:updated_at] = assessment[:updated_at] || '' if fields.include?(:updated_at)
+      if fields.include?(:symptoms)
+        symptom_names_and_labels.map(&:first).each do |symptom_name|
+          assessment_details[symptom_name.to_sym] = assessments_hash[assessment[:id]][symptom_name]
+        end
+      end
+      assessments_details << assessment_details
+    end
+    [assessments_details, symptom_names_and_labels]
+  end
+
+  # Extracts laboratory data values given relevant fields
+  def extract_laboratories_details(patients_identifiers, laboratories, fields)
+    laboratories_details = []
+    laboratories.each do |laboratory|
+      laboratory_details = {}
+      laboratory_details[:patient_id] = laboratory[:patient_id] || '' if fields.include?(:patient_id)
+      PATIENT_SECONDARY_IDENTIFIER_FIELDS.each do |identifier|
+        laboratory_details[identifier] = patients_identifiers[laboratory[:patient_id]][identifier] || '' if fields.include?(identifier)
+      end
+      laboratory_details[:id] = laboratory[:id] || '' if fields.include?(:id)
```

Comment:
Could all of these hash building bits be added as helpers right on the models (to clean this up, and might also be useful in other places)?

Reply:
good idea!

Reply:
https://github.com/SaraAlert/SaraAlert/pull/562/commits/18f2cd1c1295b23fca3eb469bf542092217c6acb#diff-7d2113c793528ea2c21a9bbc24918900a3ee6fd1bd396956de26304a9870162eR41